liberty-ls/src/main/java/io/openliberty/tools/langserver/diagnostic/LibertyPropertiesDiagnosticService.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+* Copyright (c) 2022 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation
+*******************************************************************************/
+package io.openliberty.tools.langserver.diagnostic;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+import java.util.logging.Logger;
+
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+
+import io.openliberty.tools.langserver.ls.LibertyTextDocument;
+import io.openliberty.tools.langserver.utils.PropertiesValidationResult;
+import io.openliberty.tools.langserver.utils.ServerPropertyValues;
+
+public class LibertyPropertiesDiagnosticService  {
+
+    private static final Logger LOGGER = Logger.getLogger(LibertyPropertiesDiagnosticService.class.getName());
+    private static final ResourceBundle DiagnosticMessages = ResourceBundle.getBundle("DiagnosticMessages", Locale.getDefault());
+
+
+    public Map<String, PropertiesValidationResult> compute(String text, LibertyTextDocument openedDocument) {
+        Map<String, PropertiesValidationResult> errors = new HashMap<>();
+        if (openedDocument.getUri().endsWith("bootstrap.properties") || openedDocument.getUri().endsWith("server.env")) {
+            BufferedReader br = new BufferedReader(new StringReader(text));
+            String line = null;
+            int lineNumber = 0;
+            try {
+                while ((line=br.readLine()) != null) {
+                    PropertiesValidationResult validationResult = ServerPropertyValues.validateServerProperty(line, openedDocument);
+                    if (validationResult.hasErrors()) {
+                        validationResult.setLineNumber(lineNumber);
```

Comment:
Does the IDE expect line number zero for the first line of the file?

Reply:
Yes, it is zero-indexed, different from what the user sees in the IDE line numbers.